internal/ingest/ingest_test.go

```diff
@@ -333,3 +361,19 @@ func publishRandomAdv(t *testing.T, i *legIngester, lph host.Host, lp legs.LegPu
 	}
 	return c, mhs
 }
+
+func requireTrueEventually(t *testing.T, attempt func() bool, interval time.Duration, timeout time.Duration, msgAndArgs ...interface{}) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	for {
+		if attempt() {
+			return
+		}
+		select {
+		case <-ctx.Done():
+			require.FailNow(t, "timed out awaiting eventual success", msgAndArgs...)
+			return
+		case <-time.After(interval):
```

Comment:
Create a `time.Timer` before the loop, and reset it here instead of using `time.After()` in the loop.